src/components/buttonsIndicators/button/index.tsx

```diff
@@ -21,6 +21,8 @@ export interface Props {
   onClick?: () => void
   id?: string
   disabled?: boolean
+  color?: string
+  heavy?: boolean
```

Comment:
These additions were requested by @jenn-rhim 

These are simple, non-breaking overrides that let you specify a certain color for a button regardless of its Type, and to explicitly use `font-weight: 700` for an extra bold texted button

Reply:
cc @petemill because this is completely against what we defined in language and on our meetings

Reply:
I think this warrants a discussion for the next front-end component meeting:

We have this design goal for the Welcome Page for example, and this includes certain weights and color changes that violate what is in our standard button component. I am in between a rock and a hard place trying to both respect the shared components and implement the product as delivered. It will be good to be on the same page with this.

Reply:
totally agree, design and development need to be in sync with this one, on the same page. that means everybody

Reply:
In the mean time I am going to revert the Button related portions of this PR, and we can review based on that. A follow up PR will be produced if necessary as the result of the above meeting.

Reply:
We need to add 'white' buttons, both solid and outlined, in the library for the cases where the buttons are placed on the (dark) colored background. We already have many cases in production using this style that we can benefit.

Reply:
then we need to just add it to design language and then add it to shared component as predefined type. As soon as we will add option for custom color this will be abused IMHO

Reply:
Hey @jenn-rhim @ryanml - good news: we have a proposed plan for this that hopefully solves all the requirements, and probably needs re-presenting. One of the reasons we don't want to just throw around arbitrary colors (aside from maintaining consistency and change-ability), is still supporting 'themes' e.g. for dark mode. Bad news: the plan has not yet been implemented because we're yet to be able to get buy-in from everyone and adjust the process to use theme properties properly. I'm not sure I'll do the plan justice in this comment, but I'll attempt it:

Right now we have Button variations that are defined by semantic names, that result in different colors: 
![image](https://user-images.githubusercontent.com/741836/47134967-6b624e00-d263-11e8-964e-137e182eef75.png)

The problem described here is that sometimes these buttons are on dark background and sometimes are on light background. Obviously the Default Button (which currently results in a Black button) won't look good on a dark background. The design requirement is that dark backgrounds would want different color Buttons for the same states. Right now, these darker backgrounds are used in specific known places. However, now / soon, the user will be able to make _some_ currently-light backgrounds in to dark backgrounds by changing theme preference.

So the plan is to have a concept of _theme containers_, that will know whether they are dark or light, some are always dark or light, and some will be changed as a result of user theme choice (dark or light). Essentially each part of a layout would be one of 3 things as defined by a designer:
- Always Light
- Always Dark
- Light / Dark depending on preference
(and of course the container itself would be able to have arbitrary and custom background color / graphic / whatever)

In these containers, the theme will use either Brave-Light values or Brave-Dark values depending on which theme the container is set to.

Therefore a 'Default' Button will be Black in a light container and White in a dark container. The Button would still be defined semanticly and wouldn't be able to be 'any' color over the variations that the team has defined:

| Colors        | Light-mode value           | Dark-mode value  |
| ------------- |:-------------:| -----:|
|    brandBraveInteracting | colors.orange500  |  ??? |
|    brandBraveActive | colors.orange200  |  ??? |
|    brandBraveLight | colors.orange000  |  ??? |
|     |
|    brandBatInteracting | colors.blurple500  |  ??? |
|    brandBatActive | colors.blurple200  |  ??? |
|     |
|    defaultControl | colors.grey800  | ??? |
|    defaultControlInteracting | colors.black  |  ??? |
|    defaultControlActive | colors.grey700  |  ??? |
|     |
|    warn | colors.red500  |  ??? |
|    warnInteracting | colors.red600  |  ??? |
|    warnActive | colors.red700  |  ??? |
|     |
|    subtle | colors.grey400  |  ??? |
|    subtleInteracting | colors.grey500  |  ??? |
|    subtleActive | colors.grey600  |  ??? |
    

This probably needs more of a discussion, and if this is an urgent requirement, we can hard-code a color for now, but it'd be great to ratify a solution similar to the one I've described here soon. Especially since we should be quite protective of the Button as it's an example of the rules we want to implement in other controls that allow us to reach the goals of the design system.